### PR TITLE
video: fix vittrack in the case where crop size grows until out-of-memory when the input is black

### PR DESCRIPTION
**Pull request by fengyuentau**
Fixes https://github.com/opencv/opencv/issues/25760

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [x] The PR is proposed to the proper branch
- [x] There is a reference to the original bug report and related work
- [x] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [x] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

File: modules/video/src/tracking/tracker_vit.cpp [hunk 1]
```diff
@@ -90,21 +90,26 @@ static void crop_image(const Mat& src, Mat& dst, Rect box, int factor)
...
-    dst = (dst - mean) / std;
+    int inner_size = size.width * size.height;
+    for (int i = 0; i < 3; i++) {
+        float _mean = params.meanvalue[i],
+              _std = params.stdvalue[i];
+        auto *dst_ptr = dst.ptr<float>() + i * inner_size;
+        for (int j = 0; j < inner_size; j++) {
+            dst_ptr[j] = (dst_ptr[j] - _mean) / _std;
+        }
+    }
```
Comment (asmorkalov) [inline modules/video/src/tracking/tracker_vit.cpp:112]: why not just `blobFromImageWithParams`?

**Comment by asmorkalov** [comment]
```
C:\build\precommit_windows64\4.x\opencv\modules\video\src\tracking\tracker_vit.cpp(106): warning C4244: 'initializing': conversion from 'double' to 'float', possible loss of data [C:\build\precommit_windows64\build\modules\video\opencv_video.vcxproj]
C:\build\precommit_windows64\4.x\opencv\modules\video\src\tracking\tracker_vit.cpp(107): warning C4244: 'initializing': conversion from 'double' to 'float', possible loss of data [C:\build\precommit_windows64\build\modules\video\opencv_video.vcxproj]
```

File: modules/video/src/tracking/tracker_vit.cpp (hunk 1 above)
Comment (fengyuentau) [reply modules/video/src/tracking/tracker_vit.cpp:112]: Good catch! I was not familiar with this API. Let me have a try with this.

**Review by asmorkalov — APPROVED** [review]
:+1:

**Comment by asmorkalov** [comment]
```
 [----------] 1 test from vittrack
[ RUN      ] vittrack.accuracy_vittrack
Unmatched IOU:  expect IOU val (0.642462) > the IOU threadhold (0.65)! Box 0 is [58 x 108 from (329, 88)], and Box 1 is [78 x 125 from (311, 83)]
/home/ci/opencv/modules/video/test/test_trackers.cpp:110: Failure
Value of: checkIOU(roi, targetRois[i], iouThreshold)
  Actual: false
Expected: true
Fail at img 4.
[  FAILED  ] vittrack.accuracy_vittrack (95 ms)
```

**Review by asmorkalov — APPROVED** [review]
:+1:

**Comment by asmorkalov** [comment]
Tested manually with Ubuntu 20.04.

**Pull request merged**